### ISSUE TITLE
Bundle an offline copy of mirrorlist

### ISSUE THREAD

**Issue by oldherl** [archlinux/archinstall#2598]
### describe the request

Currently archinstall gets the mirrorlist from https://archlinux.org/mirrorlist/?protocol=https , but in some cases, the user might be blocked by archlinux.org due to attackers in the neighbourhood, and gets an HTTP 403 error for that request.
![image](https://github.com/user-attachments/assets/0d2f0a17-b446-4ad6-bf9e-a81edc63f561)

Note that the user is not an attacker, but unfortunately blocked because of attackers in the same /24 or bigger subnet.

In that case, if archinstall could provide an offline copy of mirrorlist, the user could choose a mirror from that and continue the installation, as long as the chosen mirror doesn't block them.

**Comment by Torxed** [comment]
It would be a frozen mirror list that may or may not be accurate at the time of installation. But for sure, technically this would be an easy thing to implement.

In the meantime, we've adjusted some proxy configs and you should be able to access archlinux.org now :)

**Comment by oldherl** [comment]
Yes it would be a frozen list, but most mirrors in it should continue be usable within a month, which is the release cycle of archiso.

**Comment by christian-heusel** [comment]
Is `pacman-mirrorlist` not coming preinstalled with the ISO? 🤔

**Comment by Torxed** [comment]
True, that package might be there.
We're using the `archlinux.org` call to check for network connectivity to some extent.
And also while filtering/grabbing mirrors based on user selection.

**Comment by lunelovehearn** [comment]
+1 this. I think archinstall should also offer an offline, local repository to install KDE plasma, gnome, xfce and other 3 popular WMs for this case, and when the user has an unstable internet or can't access the internet at all. then, the script and the local repo gets updated monthly.
